### PR TITLE
Overlay topology survey message

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

This PR adds an http endpoint that allows a node to request connection information from another node on the network by flooding the request. This will help us understand what the current overlay topology looks like. 

A couple important details -
- This PR includes rate limiting for the flooding mechanism
- Nodes will relay/respond to messages if they are from a node in it's own transitive quorum. This behavior can be changed through the config file.
 

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: docs/software/admin.md [hunk 1]
```diff
@@ -969,6 +970,83 @@ At this point `info` will tell you that the node is setup to vote for this upgra
       ]
 ```
 
+## Overlay topology survey
```
Comment (MonsieurNicolas) [inline docs/software/admin.md]: I think this should be under `Overlay information` and not a top level section.

Also, you should link to that section from the end of `Configuration basics` with a caption `Auditing of the P2P network is enabled by default, see the "overlay topology" section for more detail if you'd like to disable it`

**Comment by MonsieurNicolas** [comment]
r+ 139a2f3e27c1535fde58afc20a004d71d2b72faf

File: src/overlay/SurveyManager.cpp [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+                             PeerStatList const& peerList) const
+{
+    for (auto const& peer : peerList)
+    {
+        Json::Value peerInfo;
+        peerInfo["nodeId"] = KeyUtils::toStrKey(peer.id);
+        peerInfo["version"] = peer.versionStr;
+        peerInfo["messagesRead"] = static_cast<Json::UInt64>(peer.messagesRead);
+        peerInfo["messagesWritten"] =
+            static_cast<Json::UInt64>(peer.messagesWritten);
```
Comment (sisuresh) [inline src/overlay/SurveyManager.cpp]: @MonsieurNicolas I had to add these casts to `Json::UInt64` to fix the build.

Comment (MonsieurNicolas) [reply src/overlay/SurveyManager.cpp]: Funny that for Json they want uint64 and not int64 (js doesn’t deal with unsigned int 64 at all)

**Comment by MonsieurNicolas** [comment]
r+ 7e2cbebac97d05ffe5314b8d491876561248cc9b

**Pull request merged**